lib/IRGen/IRGenModule.cpp

```diff
@@ -916,9 +918,9 @@ void IRGenModule::emitAutolinkInfo() {
                                        }),
                         AutolinkEntries.end());
 
-  if (TargetInfo.OutputObjectFormat == llvm::Triple::COFF ||
-      TargetInfo.OutputObjectFormat == llvm::Triple::MachO ||
-      Triple.isPS4()) {
+  if ((TargetInfo.OutputObjectFormat == llvm::Triple::COFF &&
+       (!Triple.isOSCygMing() || Triple.getArch() == llvm::Triple::thumb)) ||
```

Comment:
Lets change the test for that target.  Really, the only Windows ARM target that I know of which is tested/works is Windows itanium (with clang) or msvc (with msvc).